service/controller/cluster.go

```diff
@@ -79,7 +79,7 @@ type Cluster struct {
 	*controller.Controller
 }
 
-func NewCluster(config ClusterConfig) (*Cluster, error) {
+func NewCluster(config ClusterConfig) (*controller.Controller, error) {
```

Comment:
What do you think? Any downsides?

Reply:
Is `Cluster` struct even necessary if we go down this path?

Reply:
Yes, I forgot about it!

Reply:
Removed.

Reply:
Some controllers need to store fields like mutex that's why we do it this way.